src/optimizer/plugins/ConstructInlineInstanceOptimizer.ts

```diff
@@ -0,0 +1,95 @@
+import { pullAt, escapeRegExp } from 'lodash';
+import { OptimizerPlugin } from '../OptimizerPlugin';
+import { Package } from '../../processor';
+import {
+  ExportableAssignmentRule,
+  ExportableCaretValueRule,
+  ExportableInstance,
+  ExportableSdRule
+} from '../../exportable';
+
+export default {
+  name: 'construct_inline_instance',
+  description:
+    'Construct inline instances from groups of rules in a contained resource or a Bundle',
+
+  optimize(pkg: Package): void {
+    const inlineInstances: ExportableInstance[] = [];
+    [...pkg.instances, ...pkg.profiles].forEach(resource => {
```

Comment:
Wait, can extensions have a `^contained` element? I didn't think they inherited that.

Reply:
Extension definitions are `StructureDefinition`s -- and `StructureDefinition` has `contained` -- so you should be able to use `^contained` in an Extension definition.

Reply:
Oh `^` rules, you strike once again!

Reply:
Yeah -- last night I almost fooled myself into thinking it wasn't relevant to extensions -- mainly because they never stand on their own.  Right after I submitted my review, I almost retracted it, and then I remembered my reasoning for it, breathed a sigh of relief, cracked open a bottle of wine, lit up a fine Cuban cigar, and watched reruns of MASH all night.

Well, all of that is true except the last three bits.